client/client_test.go

```diff
@@ -1653,6 +1654,48 @@ func testTarExporterWithSocket(t *testing.T, sb integration.Sandbox) {
 	require.NoError(t, err)
 }
 
+func testTarExporterWithSocketCopy(t *testing.T, sb integration.Sandbox) {
+	if os.Getenv("TEST_DOCKERD") == "1" {
+		t.Skip("tar exporter is temporarily broken on dockerd")
+	}
+
+	requiresLinux(t)
+	c, err := New(context.TODO(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	alpine := llb.Image("docker.io/library/alpine:latest")
+	state := alpine.Run(llb.Args([]string{"sh", "-c", "nc -l -s local:/root/socket.sock & usleep 100000; kill %1"})).Root()
+
+	fa := llb.Copy(state, "/root", "/roo2",
+		&llb.CopyInfo{
+			FollowSymlinks:      true,
+			CopyDirContentsOnly: false, // !isDir
+			AttemptUnpack:       false,
+			CreateDestPath:      true,
+			AllowWildcard:       true,
+			AllowEmptyWildcard:  false,
+		})
+
+	alpine2 := alpine.File(fa)
```

Comment:
For simplicity, I guess the target state can just be `llb.Scratch()` and doesn't need to be alpine.

Reply:
nice suggestion :+1: